edgelet/edgelet-http-mgmt/src/client/module.rs

```diff
@@ -45,10 +46,10 @@ impl ModuleClient {
     }
 }
 
-fn get_base_path(url: &Url) -> &str {
+fn get_base_path(url: &Url) -> Result<PathBuf, Error> {
     match url.scheme() {
-        "unix" => url.path(),
-        _ => url.as_str(),
+        "unix" => url.to_uds_file_path().map_err(Error::from),
```

Comment:
`"unix" => Ok(url.to_uds_file_path()?),`

Reply:
On it...